### PR TITLE
Rename Accessor methods

### LINKED ISSUE #709: Rename Accessor methods after DataTable removal
* Rename accessor  read_csv
* rename accessor load_retail
    * Update statistical insights documentation to use this instead of load_retail_accessor
* rename Accessor indexers to _ilocIndexer and _locIndexer
* rename serialization files/relevant functions to just serialize and deserialize

### PR DESCRIPTION
**Pull request by jeff-hernandez**
Closes #709 by renaming accessor methods since DataTable has already been removed.

### CODE REVIEW

File: woodwork/tests/demo_tests/test_retail.py [hunk 1]
```diff
@@ -23,11 +23,11 @@ def set_testing_headers():
 
 def test_load_retail_to_accessor_diff():
```
Comment (tamargrey) [inline woodwork/tests/demo_tests/test_retail.py:24]: Saw you changed the name of the tests for `read_csv` to not include "to_accessor", which I liked. Should we do that here too?

File: woodwork/tests/demo_tests/test_retail.py [hunk 2]
```diff
@@ -36,7 +36,7 @@ def test_load_retail_to_accessor_diff():
 
 
 def test_load_retail_to_accessor():
```
Comment (tamargrey) [inline woodwork/tests/demo_tests/test_retail.py:38]: same as above--should we rename?

File: woodwork/tests/demo_tests/test_retail.py (hunk 1 above)
Comment (jeff-hernandez) [reply woodwork/tests/demo_tests/test_retail.py:24]: Yes, will update

Comment (jeff-hernandez) [reply woodwork/tests/demo_tests/test_retail.py]: ✅  bcd833a

**Review by thehomebrewnerd — APPROVED** [review]
Looks pretty good to me!

**Review by tamargrey — APPROVED** [review]
lgtm!!

**Pull request merged**